pandas/tests/plotting/test_frame.py

```diff
@@ -3229,6 +3229,20 @@ def test_subplots_sharex_false(self):
         tm.assert_numpy_array_equal(axs[0].get_xticks(), expected_ax1)
         tm.assert_numpy_array_equal(axs[1].get_xticks(), expected_ax2)
 
+    def test_plot_no_rows(self):
+        df = pd.DataFrame({"foo": [1]}, index=pd.Index(data=[], dtype=int), dtype="int")
```

Comment:
Add a reference to the GitHub issue as a comment.

This is equivalent to a `df = pd.DataFrame(columns=['foo'], dtype=int)` right?

Reply:
I'd say `pd.DataFrame(columns=['foo'], index=pd.Index(data=[], dtype=int), dtype=int)` to endorce `df.index.dtype` to be an integer dtype.

Reply:
Does it matter that the index has integer dtype?

Reply:
Indeed, the `index` parameter is actually not useful at all.